src/main/java/com/atlassian/jira/cloud/jenkins/util/BranchNameIssueKeyExtractor.java

```diff
@@ -25,29 +25,68 @@ public Set<String> extractIssueKeys(
         // The action is not injected for Pipeline (single branch) jobs
         final SCMRevisionAction scmAction = build.getAction(SCMRevisionAction.class);
 
-        if (scmAction == null) {
+        final Set<String> issueKeys = new HashSet<>();
+
+        if (scmAction != null) {
+            final SCMRevision revision = scmAction.getRevision();
+            final ScmRevision scmRevision = new ScmRevision(revision.getHead().getName());
+            issueKeys.addAll(extractIssueKeys(scmRevision.getHead()));
+
             pipelineLogger.debug(
                     String.format(
-                            "Could not extract issue keys from branch name of build %s due to: SCMRevisionAction is null",
-                            build.number));
-            return Collections.emptySet();
+                            "Extracted issue keys from scmRevision.getHead() (%s): %s",
+                            scmRevision.getHead(), issueKeys));
+        } else {
+            pipelineLogger.debug(
+                    "Not extracting issue keys from scmRevision.getHead() because it's not set");
         }
 
-        final SCMRevision revision = scmAction.getRevision();
-        final ScmRevision scmRevision = new ScmRevision(revision.getHead().getName());
+        final Map<String, String> envVars = build.getEnvVars();
+
+        // You can get an overview of a Jenkins server's environment variables by opening
+        // http://your-jenkins-server/env-vars.html
 
-        Set<String> issueKeys = extractIssueKeys(scmRevision);
+        // For a multibranch project, this will be set to the name of the branch being built, for example in case you
+        // wish to deploy to production from master but not from feature branches; if corresponding to some kind of
+        // change request, the name is generally arbitrary
+        final String branchNameEnvVar = envVars.get("BRANCH_NAME");
+        if (branchNameEnvVar != null) {
+            issueKeys.addAll(extractIssueKeys(branchNameEnvVar));
+            pipelineLogger.debug(
+                    String.format(
+                            "Extracted issue keys from env var BRANCH_NAME (%s): %s",
+                            branchNameEnvVar, issueKeys));
```

Comment:
/nit, `issueKeys` contains the keys extracted from previous `scmRevision.getHead()`, so the log message could be a bit ambiguous.

Reply:
Good catch! Will fix this.